test/AutoinstConfig_tests.rb

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "AutoinstConfig"
+
+describe Yast::AutoinstConfig do
+  subject { Yast::AutoinstConfig }
+
+  describe "#find_slp_autoyast" do
+    before do
+      allow(Yast::SLP).to receive(:FindSrvs).with("autoyast", "").and_return(slp_server_reply)
+    end
+
+    context "when no 'autoyast' provider returned by SLP" do
+      let(:slp_server_reply) { [] }
+
+      it "returns nil" do
+        expect(subject.find_slp_autoyast).to eq(nil)
+      end
+    end
+
+    context "when only one 'autoyast' provider returned by SLP" do
+      let(:service_url) { "https://192.168.0.1/autoinst.xml" }
+      let(:slp_server_reply) { [{"srvurl" => "service:autoyast:#{service_url}" }] }
```

Comment:
NP: Someday we should try to enable Rubocop in yast2-autoinstallation :) (just missing an extra space between `{` and `"`.